paginate/src/main/java/com/paginate/recycler/WrapperAdapter.java

```diff
@@ -66,7 +66,11 @@ boolean isDisplayLoadingRow() {
     void displayLoadingRow(boolean displayLoadingRow) {
         if (this.displayLoadingRow != displayLoadingRow) {
             this.displayLoadingRow = displayLoadingRow;
-            notifyDataSetChanged();
+            if(displayLoadingRow){
+                notifyItemInserted(wrappedAdapter.getItemCount());
```

Comment:
This actually doesn't need to be true. Most of the times items will be added at the end of the list but it doesn't need to be the case. With this line you are saying that items are always inserted at the end of the list.